enterprise/coderd/idpsync.go

```diff
@@ -363,3 +363,58 @@ func (api *API) idpSyncClaimFields(orgID uuid.UUID, rw http.ResponseWriter, r *h
 
 	httpapi.Write(ctx, rw, http.StatusOK, fields)
 }
+
+// @Summary Get the organization idp sync claim field values
+// @ID get-the-organization-idp-sync-claim-field-values
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param organization path string true "Organization ID" format(uuid)
+// @Param claimField query string true "Claim Field" format(string)
+// @Success 200 {array} string
+// @Router /organizations/{organization}/settings/idpsync/field-values [get]
+func (api *API) organizationIDPSyncClaimFieldValues(rw http.ResponseWriter, r *http.Request) {
+	org := httpmw.OrganizationParam(r)
+	claimField := r.URL.Query().Get("claimField")
+	api.idpSyncClaimFieldValues(org.ID, claimField, rw, r)
+}
+
+// @Summary Get the idp sync claim field values
+// @ID get-the-idp-sync-claim-field-values
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param organization path string true "Organization ID" format(uuid)
+// @Param claimField query string true "Claim Field" format(string)
+// @Success 200 {array} string
+// @Router /settings/idpsync/field-values [get]
+func (api *API) deploymentIDPSyncClaimFieldValues(rw http.ResponseWriter, r *http.Request) {
+	claimField := r.URL.Query().Get("claimField")
+	// nil uuid implies all organizations
+	api.idpSyncClaimFieldValues(uuid.Nil, claimField, rw, r)
+}
+
+func (api *API) idpSyncClaimFieldValues(orgID uuid.UUID, claimField string, rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	fields, err := api.Database.OIDCClaimFieldValues(ctx, database.OIDCClaimFieldValuesParams{
```

Comment:
just do the `claimField` parsing in here.
```suggestion
	ctx := r.Context()

        claimField := r.URL.Query().Get("claimField")
	fields, err := api.Database.OIDCClaimFieldValues(ctx, database.OIDCClaimFieldValuesParams{
```

Also you should defend against an empty string here.

```
if claimField == "" {
		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
			Message: "claimField query parameter is required",
		})
		return
}
```